operator/runtime/stagenet/src/configs/mod.rs

```diff
@@ -1513,6 +1503,46 @@ impl pallet_external_validators_rewards::types::SendMessage for RewardsSendAdapt
     }
 }
 
+/// Wrapper to check if a validator sent a heartbeat in the current session
+/// Note: This only checks for actual heartbeats, not block production
+pub struct ValidatorIsOnline;
+impl frame_support::traits::Contains<AccountId> for ValidatorIsOnline {
+    fn contains(account: &AccountId) -> bool {
+        let validators = Session::validators();
+        if let Some(index) = validators.iter().position(|v| v == account) {
+            // Check if this validator sent a heartbeat (not just produced blocks)
+            ImOnline::received_heartbeat_in_current_session(index as u32)
+        } else {
+            // Not a validator in current session, consider offline
+            false
+        }
+    }
+}
+
+/// Wrapper to check if a validator has been slashed in a given era
+pub struct ValidatorSlashChecker;
+impl pallet_external_validators_rewards::SlashingCheck<AccountId> for ValidatorSlashChecker {
+    fn is_slashed(era_index: u32, validator: &AccountId) -> bool {
+        pallet_external_validator_slashes::ValidatorSlashInEra::<Runtime>::contains_key(
+            era_index, validator,
+        )
+    }
+}
+
+parameter_types! {
+    /// Expected number of blocks per era for inflation scaling
+    /// Calculation: SessionsPerEra × EpochDurationInBlocks
+    /// - Production: 6 sessions × 600 blocks = 3600 blocks (6 hours at 6s/block)
+    /// - Fast runtime: 1 session × 10 blocks = 10 blocks (for testing)
+    pub const ExpectedBlocksPerEra: u32 = polkadot_runtime_common::prod_or_fast!(3600, 10);
```

Comment:
We should use the constants instead of the hardcoded values
```suggestion
    /// Expected number of blocks per era for inflation scaling
    /// Calculation: SessionsPerEra × EpochDurationInBlocks
    pub const ExpectedBlocksPerEra: u32 =
        (SessionsPerEra::get() as u32).saturating_mul(EpochDurationInBlocks::get() as u32);
```